cocos/ui/scroll-view.ts

```diff
@@ -857,7 +787,11 @@ export class ScrollView extends ViewGroup {
      * ```
      */
     public scrollToPercentVertical (percent: number, timeInSecond?: number, attenuated?: boolean): void {
-        assignMoveDeltaOption(0, percent, false, true);
+        this._doScroll(0, percent, false, true, timeInSecond, attenuated);
+    }
+
+    private _doScroll (x: number, y: number, applyToHorizontal: boolean, applyToVertical: boolean, timeInSecond?: number, attenuated = true): void {
+        assignMoveDeltaOption(x, y, applyToHorizontal, applyToVertical);
```

Comment:
The function assignMoveDeltaOption can be removed?

Reply:
No, it will update `_moveDeltaOptions`

Reply:
I mean to put the implementation of assignMoveDeltaOption directly here?

Reply:
It is better not. Current implementation make it more clearly.